### ISSUE TITLE
Allow `opportunities` resource search to operate asynchronously

### ISSUE THREAD

**Issue by nic-planet** [stapi-spec/stapi-spec#77]
Computing opportunities can take a bit too long to be handled by a synchronous request. In the synchronous request for the sample backend, at least for the Planet API, we in fact encapsulate an asynchronous request and only return once it's finished. Also, if a customer polls results from the async opportunity collection process and only a part of the results have been gathered, it would be good if those results would already be shown, along with information how to get the following ones (basically pagination, but not creating page breaks based on the number of results, but on the time of creation). See https://developers.planet.com/docs/tasking/examples/assured/ for one way to expose this kind of functionality, in particular the "next" field that would display the next set of results (even though unused in the example).

**Comment by matthewhanson** [comment]
Should this be required behavior of the API (handling async requests), if you implement the /opportunities endpoint?

If async this implies there needs to be some sort of Opportunities request ID.  What other fields would be needed?  

How would the partial results work in conjunction with pagination as well if pages were needed?

**Comment by nic-planet** [comment]
Yes, I think it would be good to expose this to users, not wrap it in the individual adaptor - it allows to return some quick results in the near future for quick feedback to the users while longer-running operations for further out dates are still ongoing. Wrapping it would mean we have to wait for everything to conclude to make it synchronous again. 

And yes, it's very related to paging (one might say it's a kind of paging where the page break is determined by arrival time of the data instead of page size). Maybe we could find a formulation to cover both under a shared concept of paging.

**Comment by matthewhanson** [comment]
Things that need to be done
- Update products with parameter to indicate sync/async per product
- Update OpenAPI docs to allow for async
- Add endpoints for listing previous opportunities requests

Investigate OGC Processes API, which handles async, to see if we could use that structure
